frontend/packages/ceph-storage-plugin/src/components/create-storage-system/reducer.ts

```diff
@@ -125,12 +121,80 @@ export type LocalVolumeSet = {
   chartNodes: Set<string>;
 };
 
+const setDeployment = (
+  prevState: WizardState,
+  newState: WizardState,
+  deploymentType: DeploymentType,
+) => {
+  /*
+   * Wizard state should be reset when a new deployment type is selected
+   * in order to avoid stale state collisions since each deployment mode
+   * has its own supported configuration.
+   *
+   * Its not required if the user has not visited more than first step.
+   */
+  if (prevState.stepIdReached !== 1) {
+    const { type } = prevState.backingStorage;
+    return {
+      ...initialState,
+      storageClass:
+        type === BackingStorageType.EXISTING ? prevState.storageClass : initialState.storageClass,
+      backingStorage: {
+        ...prevState.backingStorage,
+        deployment: deploymentType,
+      },
+    };
+  }
```

Comment:
Is this logic really required?

Reply:
Yes its required and I added comment

Reply:
it says it's required for step 1 pages only why can't we reset the state for all cases. What difference would it make? Could you please elaborate the comment was not clear enough hence I have added this comment.

Reply:
It saying not req for step 1 =
`Its not required if the user has not visited more than first step` -> more than a single step.
```
  /*
   * Wizard state should be reset when a new backing storage type is selected
   * in order to avoid stale state collisions since each backing storage type
   * has its own supported variables. e.g if arbiter was selected and not
   * deselected before changing the backing storage type then storage cluster spec
   * will mark the arbiter option as enabled.
   *
   * Its not required if the user has not visited more than first step.
   */
   ```

   Its to avoid stale states, every mode of deployment sets some options in global state, if that mode f deployment is dropped it should be reset , other wise previous global state will be overwritten.
  Visually you will have steps visited in wizard nav even though user did not actually went through those.
  And code wise you can have kms enabled , arbiter network options enabled in diff modes where they are not supported and you will have a bad stoarge cluster yaml. So a couple of bad effects if not reset.
  This is all in 4.9 too , I just isolated it here so it will be easy to follow .

Reply:
Therefore for first step , its not needed to reset whole global state, since user has not visited more steps.
just doing when there are steps visited and then deselection happening